### PR TITLE
CI: Refactor Sonatype Nexus IQ workflow

### PR DESCRIPTION
**Pull request by jwagantall**
- Gradle build is now a separate action
- This will allow to run all tasks in 1 job

### CODE REVIEW

File: .github/actions/gradle-build-action/action.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+inputs:
+  jdk-version:
+    description: "OpenJDK version"
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - id: Setup JDK ${{ inputs.jdk-version }}
+        uses: actions/setup-java@v3
```
Comment (MightyNerdEric) [inline .github/actions/gradle-build-action/action.yaml:14]: Bad intention here.

File: .github/workflows/composed-gradle-nexus-iq.yaml [hunk 2]
```diff
@@ -62,25 +62,27 @@ env:
...
+          gerrit-url: ${{ vars.GERRIT_URL }}
+          delay: "0s"
+          submodules: "true"
+      # yamllint disable-line rule:line-length
+      - uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c  # v5.0.0
+        id: setup-python
+        with:
+          python-version: "3.8"
+      - name: Run gradle build and CLM scan
+        uses: lfit/releng-reusable-workflows/.github/actions/gradle-build-action@main
```
Comment (MightyNerdEric) [inline .github/workflows/composed-gradle-nexus-iq.yaml:83]: This needs a yamllint line-length exemption.

**Review by MightyNerdEric — CHANGES_REQUESTED** [review]

File: .github/actions/gradle-build-action/action.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        with:
+          java-version: ${{ inputs.jdk-version }}
+          distribution: 'oracle'
```
Comment (zxiiro) [inline .github/actions/gradle-build-action/action.yaml:17]: Oracle? Should we be using OpenJDK instead?

Comment (zxiiro) [reply .github/actions/gradle-build-action/action.yaml:17]: `temurin` is the Eclipse Foundation's OpenJDK prebuilt package.

File: .github/actions/gradle-build-action/action.yaml (hunk 1 above)
Comment (jwagantall) [reply .github/actions/gradle-build-action/action.yaml]: Thank you! you are right..

File: .github/workflows/composed-gradle-nexus-iq.yaml (hunk 2 above)
Comment (jwagantall) [reply .github/workflows/composed-gradle-nexus-iq.yaml:83]: added

File: .github/actions/gradle-build-action/action.yaml (hunk 3 above)
Comment (jwagantall) [reply .github/actions/gradle-build-action/action.yaml]: thank you! let me switch that

**Review by zxiiro — APPROVED** [review]

**Review by MightyNerdEric — APPROVED** [review]

**Pull request merged**